### PR TITLE
[Messenger] add dedicated method for disabling instead of passing boolean flags

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 6.2
| Bug fix?      | no
| New feature?  | yes (changes #39622)
| Deprecations? | no
| Tickets       | 
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Review by chalasr — APPROVED** [review]

**Comment by xabbuh** [comment]
ping @lyrixx What do you think as you introduced the initial method (in #39622)?

**Comment by lyrixx** [comment]
I Don't have any recommendations here... It's up to you 👍🏼

File: src/Symfony/Component/Messenger/Command/FailedMessagesRemoveCommand.php [hunk 1]
```diff
@@ -78,7 +78,7 @@ private function removeMessages(string $failureTransportName, array $ids, Receiv
             try {
                 $envelope = $receiver->find($id);
             } finally {
-                $this->phpSerializer?->enableClassNotFoundCreation(false);
+                $this->phpSerializer?->disableClassNotFoundCreation();
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Messenger/Command/FailedMessagesRemoveCommand.php]: what about allowPhpIncompleteClass? that's even more explicit I think

File: src/Symfony/Component/Messenger/Transport/Serialization/PhpSerializer.php [hunk 2]
```diff
@@ -23,9 +23,14 @@ class PhpSerializer implements SerializerInterface
 {
     private bool $createClassNotFound = false;
 
-    public function enableClassNotFoundCreation(bool $enable = true): void
+    public function enableClassNotFoundCreation(): void
```
Comment (lyrixx) [inline src/Symfony/Component/Messenger/Transport/Serialization/PhpSerializer.php]: Actually, I would add `@internal` on this method, whatever the name is!

File: src/Symfony/Component/Messenger/Command/FailedMessagesRemoveCommand.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Symfony/Component/Messenger/Command/FailedMessagesRemoveCommand.php]: acceptPhpIncompleteClass / rejectPhpIncompleteClass ?

Comment (xabbuh) [reply src/Symfony/Component/Messenger/Command/FailedMessagesRemoveCommand.php]: Sounds good 👍 I’ll update the PR tonight

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**